modules/launcher/src/main/java/org/mule/module/launcher/DefaultArchiveDeployer.java

```diff
@@ -550,6 +468,7 @@ public void redeploy(T artifact) throws DeploymentException
             try
             {
                 logDeploymentFailure(t, artifact.getArtifactName());
+                addZombieApp(artifact);
```

Comment:
Here the artifact is added as zombie when the redeployment has failed. It will be removed from the zombie map on the next successful redeployment.